lib/reek/context/code_context.rb

```diff
@@ -12,10 +16,13 @@ module Context
     # :reek:TooManyMethods: { max_methods: 19 }
     # :reek:TooManyInstanceVariables: { max_instance_variables: 8 }
     class CodeContext
-      attr_reader :exp
-      attr_reader :num_statements
-      attr_reader :children
-      attr_reader :visibility
+      extend Forwardable
+      def_delegators :exp, :name, :type
+      def_delegators :visibility_tracker, :visibility, :visibility=, :non_public_visibility?
```

Comment:
`def_delegators`’ first argument always trips me up, so I have a slight preference for the more explicit `delegate` syntax:

```ruby
      delegate %i(name type) => :exp
      delegate %i(visibility visibility= non_public_visibility?) => :visibility_tracker
```

WDYT?